### ISSUE TITLE
Detached Thread Without Lifetime Safety Guarantees

### ISSUE THREAD

**Issue by jwinarske** [toyota-connected/plugins#231]
Affected File: [plugins/flatpak/flatpak_shim.cc](https://github.com/toyota-connected/ivi-homescreen-plugins/blob/9c2063d4512cb18b308f440bd245cf21d4e75096/plugins/flatpak/flatpak_shim.cc#L1111-L1113)

toyota-connected / ivi-homescreen-plugins / plugins / flatpak / flatpak_shim.cc
```C++
std::thread([self, transaction_raw, callback, ref_name, remote_name,
             strand_ptr = strand_, installation_raw, found_ref_raw, id]() {
  // ...
  std::this_thread::sleep_for(std::chrono::seconds(2));  // ← hardcoded race
```

Risk:

A hardcoded 2-second sleep is used as a heuristic to wait for filesystem sync — this is a time-of-check/time-of-use (TOCTOU) race condition pattern. On a slow system, the filesystem may not be synced; on a fast system, it's an unnecessary delay.
The std::thread is immediately detached (.detach() implicit via lambda destruction path), meaning if the plugin object is destroyed before the thread completes, the captured self (shared_from_this) reference is the only thing keeping the object alive. If the callback or strand is invalidated, this can lead to use-after-free.
transaction_raw, installation_raw, found_ref_raw are raw GObject* pointers whose ownership semantics in the detached thread are unclear.
Recommendation:

Replace the sleep with a proper inotify or GFile monitor for filesystem events.
Use a std::future/std::promise pair and proper thread join semantics.
Audit all captured raw GObject* pointers for ownership correctness.
